manifests/init.pp

```diff
@@ -27,43 +27,27 @@
 # Copyright 2014 Rob Nelson
 #
 define local_user (
-  $state,
-  $groups,
-  $password,
-  $comment             = $name,
-  $uid                 = undef,
-  $gid                 = $name,
-  $last_change         = 0,
-  $password_max_age    = 90,
-  $shell               = '/bin/bash',
-  $home                = "/home/${name}",
-  $managehome          = true,
-  $ssh_authorized_keys = [],
-  $manage_groups       = false,
-  $system              = false,
+  Enum['present','absent'] $state,
+  Array $groups,
```

Comment:
I guess this array should only contain strings? You can do:
Array[String] $groups,